mozci/util/taskcluster.py

```diff
@@ -86,6 +87,7 @@ def find_task_id(index_path, use_proxy=False):
         response = _do_request(get_index_url(index_path))
     except requests.exceptions.HTTPError as e:
         if e.response.status_code == 404:
-            raise KeyError("index path {} not found".format(index_path))
+            logging.getLogger(__name__).warning("index path {} not found".format(index_path))
+            return None
```

Comment:
I think it's better to raise here and let consumers handle the exception. Maybe that's in the `push.py` file, or maybe it's in the `scheduler_analysis` script. I'd need to think about it.

Reply:
That is up to you.  I do not intend for this PR to be the real solution.  Just a pointer to what needs to be handled.  How you handle it is up to you.

That said `find_task_id()` should return `None` when the task id does not exists.  Which is the case here.